fvm/state/state_test.go

```diff
@@ -6,48 +6,109 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/fvm/utils"
 )
 
-func TestState_DraftFunctionality(t *testing.T) {
-	ledger := state.NewMapLedger()
-	st := state.NewState(ledger)
+func TestState_ChildMergeFunctionality(t *testing.T) {
+	view := utils.NewSimpleView()
+	st := state.NewState(view)
+
+	t.Run("test read from parent state (backoff)", func(t *testing.T) {
+		key := "key1"
+		value := createByteArray(1)
+		// set key1 on parent
+		err := st.Set("address", "controller", key, value)
+		require.NoError(t, err)
+
+		// read key1 on child
+		stChild := st.NewChild()
+		v, err := stChild.Get("address", "controller", key)
+		require.NoError(t, err)
+		require.Equal(t, v, value)
+	})
+
+	t.Run("test write to child (no merge)", func(t *testing.T) {
+		key := "key2"
+		value := createByteArray(2)
+		stChild := st.NewChild()
+
+		// set key2 on child
+		err := stChild.Set("address", "controller", key, value)
+		require.NoError(t, err)
+
```

Comment:
shouldn't we also check reading the same key on the parent?

Reply:
added